docs/application/native/guides/text-input/autofill.md

```diff
@@ -0,0 +1,365 @@
+
+- [Requesting to store the data entered by the user in autofill service](#requesting-to-store-the-data-entered-by-the-user-in-autofill-service)
+
+  Use the `autofill_commit()` function to store the data entered by the user in autofill service
+
+## Prerequisites
+
+1. Configure autofill service:
+
+   Set autofill service in Settings > Language and input > Input Assistance > Autofill setting.
+   In some environment, the autofill service can be designated according to the system configuration.
+
+2. To use the autofill API (in [mobile](../../api/mobile/latest/group__CAPI__UIX__AUTOFILL__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__UIX__AUTOFILL__MODULE.html) applications), include the `<autofill.h>` header file in your application:
+
+   ```c
+   #include <autofill.h>
+   ```
+
+3. To use the autofill library, create an autofill handle:
+
+   The autofill handle is used in other autofill functions as a parameter.
+
+   ```c
+   void
+   create_autofill_handle()
+   {
+       autofill_h ah;
+       int ret;
+       ret = autofill_create(&ah);
+       if (ret != AUTOFILL_ERROR_NONE)
+           /* Error handling */
+   }
+   ```
+
+4. When you no longer need the autofill library, destroy the autofill handle using the `autofill_destroy()` function:
+
+   ```c
+   void
+   destroy_autofill_handle(autofill_h ah)
+   {
+       int ret;
+
+       ret = autofill_destroy(ah); /* ah is the autofill handle */
+       if (ret != AUTOFILL_ERROR_NONEt)
+           /* Error handling */
+   }
+   ```
+
+   > **Note**
+   >
+   > Do not use the `autofill_destroy()` function in a callback function.
+
+5. After you create the autofill handle, connect the background autofill daemon with the `autofill_connect()` function:
+   The function is asynchronous and you can get the result with callback function that registered as the second parameter of `autofill_connect()`.
+
+   ```c
+   static void connection_status_changed_cb(autofill_h ah, autofill_connection_status_e status, void *user_data)
+   {
+       switch (status) {
+           case AUTOFILL_CONNECTION_STATUS_CONNECTED:
+               dlog_print(DLOG_INFO, LOG_TAG, "connected");
+               break;
+           case AUTOFILL_CONNECTION_STATUS_DISCONNECTED:
+               dlog_print(DLOG_INFO, LOG_TAG, "disconnected");
+               break;
+           case AUTOFILL_CONNECTION_STATUS_REJECTED:
+               dlog_print(DLOG_INFO, LOG_TAG, "rejected");
+               break;
+           default:
+               break;
+   }
+
+   void
+   connect_autofill_daemon(autofill_h amh)
+   {
+       int ret;
+       ret = autofill_connect(amh, connection_status_changed_cb, NULL);
+       if (ret != AUTOFILL_ERROR_NONEt)
+           /* Error handling */
+   }
+   ```
+
+   You can use the autofill functions after receiving `AUTOFILL_CONNECTION_STATUS_CONNECTED` in `connection_status_changed_cb()`.
```

Comment:
You can use the Autofill functions after receiving the status from  AUTOFILL_CONNECTION_STATUS_CONNECTED in the `connection_status_changed_cb()` function.